internal/config/ca_config/ca_config.go

```diff
@@ -0,0 +1,50 @@
+package ca_config
+
+type CAIdType int
+
+const (
+	InternalCA CAIdType = iota + 1
+	AsyncInternalCA
+)
+
+type InternalCACfg struct {
+	CaCertFile         string `json:"caCertFile,omitempty"`
+	CaKeyFile          string `json:"caKeyFile,omitempty"`
+	SignerCertName     string `json:"SignerCertName,omitempty"`
+	CaSerialFile       string `json:"caSerialFile,omitempty"`
+	CaCertValidityDays int    `json:"CaCertValidityDays,omitempty"`
+	CACertStore        string `json:"CACertStore,omitempty"`
+}
+
+type CAConfigType struct {
+	CAType                          CAIdType       `json:"CAType,omitempty"`
+	AdminCommonName                 string         `json:"AdminCommonName,omitempty"`
+	ClientBootstrapCommonName       string         `json:"ClientBootstrapCommonName,omitempty"`
+	ClientBootstrapCertName         string         `json:"ClientBootstrapCertName,omitempty"`
+	ClientBootstrapCommonNamePrefix string         `json:"ClientBootstrapCommonNamePrefix,omitempty"`
```

Comment:
Do we want to allow users to set these via the config file?
Downside is it increases our config surface, meaning more testing and more constraints if we need to make changes later.

Reply:
Either that or define them as global constants. They were repeated in quite a few places.